### PR TITLE
Improve ISSUE_TEMPLATE.md

### PR DESCRIPTION
**Pull request by fmachado091**
- Fixes typos
- Organizes sections
- Improves formatting and word phrasing
- Removes ADR reference in favor of RFC

Link to the file: [Current](https://github.com/quintoandar/.github/blob/69b5382750ffd799de863133bbc2736d03eb62c7/.github/ISSUE_TEMPLATE.md) vs [Proposed](https://github.com/quintoandar/.github/blob/35c8b3ccdcd9c5d7b167533dfdce03b82f4369b2/.github/ISSUE_TEMPLATE.md)

### CODE REVIEW

**Review by lucianohgo — APPROVED** [review]
Cool :) Might be good to have somewhere a way to add what jira ticket this is related to. Could be optional since not all teams use Jira

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -4,71 +4,56 @@ labels: 'tech-debt'
...
+Give a clear and concise description of what the tech debt is.
 
-### Solution Hints :shipit:
+## Impact :bomb:
 
-<!--
-Here you should add only hints, context, and opinions
-but let the solution proposal in a PR for ADR.
--->
+Include a description of the current or possible impact of this tech debt and when it's estimated to become critical.
```
Comment (danielasy) [inline .github/ISSUE_TEMPLATE.md:47]: Maybe we can keep the new hint texts as comments, just like the label explanation at the top, so people won't have to delete it when filling in the sections

```suggestion
<!--
Include a description of the current or possible impact of this tech debt
and when it's estimated to become critical.
-->
```

**Review by danielasy — APPROVED** [review]

**Review by bielcapanema — APPROVED** [review]

**Review by woliveiras — APPROVED** [review]
Lança!

**Comment by fmachado091** [comment]
@lucianohgo 

> Cool :) Might be good to have somewhere a way to add what jira ticket this is related to. Could be optional since not all teams use Jira

Do you think it could be in the "further info" section?

**Pull request merged**